### DISCUSSION TITLE
Visual block mode

### DISCUSSION THREAD

**Discussion by F0Xde** [helix-editor/helix#1824] [Ideas] [answered]
Is there currently anything similar to Vim's visual block mode (`<C-v>`), and if not could this implemented in the future?

**Comment by sudormrfbin** [comment, marked as answer]
`C` and `Alt-C` adds additional cursors below and above the primary cursor, and all subsequent movement and actions work on all cursors.

**Reply by F0Xde** [reply]
Oh true I forgot multiple cursors can be used that way in combination with visual mode. 
What I noticed is that this doesn't work as expected when some lines are shorter than others, as the selection will then skip to the start of the next line. Additionally it may be a bit nicer to have a real visual block mode that can be used like the standard visual mode, meaning with a single cursor and using j/k.

**Reply by parisni** [reply]
> when some lines are shorter than others, as the selection will then skip to the start of the next line.

turns out, when you press `gl` (go to line end) then the selection won't have the behavior. You will get every lines whenever one is shorter.

**Comment by kallisti5** [comment]
<3 visual block mode.  I think there are some other vi emulations which might be more critical for basic usage, but visual block mode is *really* handy to mass-indent, etc.

**Comment by kallisti5** [comment]
I was pointed to https://github.com/LGUG2Z/helix-vim/blob/master/config.toml by djo_kirawi on matrix.  It helps somewhat.  Visual Block mode isn't really working though.

**Comment by AK-10** [comment]
How can i exit multiple cursors?

I'd like to exit `esc` or `ctrl-[`.

**Reply by Proful** [reply]
@AK-10 Use , (comma)

I prefer ESC though

**Reply by AK-10** [reply]
Oh, thank you!

**Reply by yonatan8070** [reply]
You can bind ESC to do that and remove the selection like so:
```toml
[keys.normal]
"esc" = ["keep_primary_selection", "collapse_selection"]
```
